.github/workflows/staging_preview.yaml

```diff
@@ -0,0 +1,60 @@
+# Only run for PR with modification to staging.
+on:
+  pull_request:
+    paths: 
+      - R/staging.txt
+  
+name: Update staging infos
+jobs:
+  create_md:
+    name: fetch updated info
+    runs-on: ubuntu-20.04
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      RENV_PATHS_ROOT: ~/.local/share/renv
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # checkout PR branch or push branch
+          ref: ${{ github.head_ref }}
+      
+      - uses: r-lib/actions/setup-r@v1
+        id: install-r
+      
+      - name: Cache Renv packages
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.RENV_PATHS_ROOT }}
+          key: ${{ runner.os }}-renv-1-${{ hashFiles('R/renv.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-renv-1-
+            ${{ runner.os }}-renv-
+            
+      - name: Install packages
+        run: | 
+          if (!requireNamespace("renv", quietly = TRUE)) install.packages("renv")
+          renv::restore(repos = "https://packagemanager.rstudio.com/all/latest")
+        shell: Rscript {0}
+        working-directory: R/
+      
+      # create-md.R only create the file for stating.txt url
+      - name: Update books
+        env:
+          RSC_BOOKDOWN_ORG_TOKEN: ${{ secrets.RSC_BOOKDOWN_ORG_TOKEN }}
+          PINS_VERBOSE: true
+        run: Rscript create-md.R
+        working-directory: R/
+        
+      - name: move to external.txt
+        run: |
+          cat staging.txt >> external.txt
+          cat /dev/null > staging.txt
+        working-directory: R/
+        
+      - name: commit
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+          git add content/\* R/\*
+          git commit -m 'new book infos'
+          git push
```

Comment:
Just want to make sure I understand it correctly: this will push to the branch associated with the PR, right? If a user submits a PR from a branch in their forked repo, this should push to the branch in the forked repo, and hence the PR will be updated automatically.

Reply:
Yes exactly. 

This works like that because i did checkout the head_ref from the PR and not the merge commit (which is the default behavior)

https://github.com/rstudio/bookdown.org/blob/53b08e9c17013f2e47266428bf2b6c80bb30476e/.github/workflows/staging_preview.yaml#L15-L19

This means : 

* the branch of the PR is checkout
* All the task are happening in the Github workspace which is in the state of this branch
* Pushing with git will push to this branch. 

I believe this will work from a fork. I did not test though... 
Do you have a new book to add ? 😃

Reply:
Perfect!

> Do you have a new book to add ?

I just took a look at the bookdown tweets at https://t.yihui.org/post/2021/03/07/ and found https://ykunisato.github.io/jpa2020-tws-002/. You may use it for testing.

Reply:
yep you were right to ask : this is not working #79 

I'll look into that. 

I have one solution that would involve to rely on an issue comment to validate the commit. 
Inspired by: https://github.com/r-lib/actions/tree/master/examples#commands-workflow
which is working because jim developed special action to fetch from and push to PRs.